Projects/TestProject/TestTransactionAddRelation.cs

```diff
@@ -0,0 +1,87 @@
+﻿using Microsoft.VisualStudio.TestTools.UnitTesting;
+using BackendlessAPI;
+using BackendlessAPI.Persistence;
+using System;
+using System.Collections.Generic;
+using BackendlessAPI.Transaction;
+using Weborb.Writer;
+using System.Reflection.PortableExecutable;
+
+namespace TestProject
+{
+  [TestClass]
+  public class TestTransactionAddRelation
+  {
+    [TestMethod]
+    public void TestAddRelationDictionary()
+    {
+      UnitOfWork unitOfWork = new UnitOfWork();
+
+      String[] giftId = { (String) Backendless.Data.Of( "Order" ).FindFirst()[ "objectId" ] };
+
+      String parentObjectId = (String) Backendless.Data.Of( "Person" ).FindById( "8B329DB4-46DB-03A1-FF1A-6828DF619400" )[ "objectId" ];
+      String parentTableName = "Person";
+      String relationColumnName = "Surname";
+
+      unitOfWork.AddToRelation( parentTableName, parentObjectId, relationColumnName, giftId );
+
+      UnitOfWorkResult result = unitOfWork.Execute();
+      Assert.IsTrue( ( (Dictionary<Object, Object>) result.Results[ "addToRelationPerson1" ] ).Count == 2 );
+    }
+
+    [TestMethod]
+    public void TestAddRelationClass()
+    {
+      UnitOfWork unitOfWork = new UnitOfWork();
+
+      List<Order> gifts = new List<Order>();
+
+      Order order = new Order();
+      order.SetObjectId( (String) Backendless.Data.Of( "Order" ).FindFirst()[ "objectId" ] );
```

Comment:
no, this is wrong. Use the object which you retrieve from the server. Do not create a new object and do not set objectId